build-cake.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# set the DOTNET_REFERENCE_ASSEMBLIES_PATH to mono reference assemblies folder
+# https://github.com/dotnet/cli/issues/531
+if [ -z "$DOTNET_REFERENCE_ASSEMBLIES_PATH" ]; then
+    if [ $(uname) == Darwin ] && [ -d "/Library/Frameworks/Mono.framework/Versions/Current/lib/mono/xbuild-frameworks" ]; then
+        export DOTNET_REFERENCE_ASSEMBLIES_PATH="/Library/Frameworks/Mono.framework/Versions/Current/lib/mono/xbuild-frameworks"
+    elif [ -d "/usr/local/lib/mono/xbuild-frameworks" ]; then
+        export DOTNET_REFERENCE_ASSEMBLIES_PATH="/usr/local/lib/mono/xbuild-frameworks"
+    elif [ -d "/usr/lib/mono/xbuild-frameworks" ]; then
+        export DOTNET_REFERENCE_ASSEMBLIES_PATH="/usr/lib/mono/xbuild-frameworks"
+    fi
+fi
+
+if [ $(uname) == Darwin ]; then
+    export OSSTRING=osx
+elif [ $(uname) == Linux ]; then
+    export OSSTRING=linux
+fi
+#curl -Lsfo cake-bootstrap.sh http://cakebuild.net/bootstrapper/$OSSTRING
+#chmod +x cake-bootstrap.sh
+./cake-bootstrap.sh "$@"
```

Comment:
I was blocked on https://github.com/cake-build/cake/issues/737 and https://github.com/cake-build/cake/pull/734. The latter was merged, but not yet released (at least not on GitHub). I created a PR (https://github.com/cake-build/cake/pull/756) to fix the former. Another option is to hard-code the NUGET_EXE environment variable in this script.

Long term PR https://github.com/cake-build/cake/pull/605 would also be beneficial, as custom arguments are not allowed otherwise (just learned this today).